File: src/test/java/org/mastodon/mamut/TestingAdjustableSlider.java
package org.mastodon.mamut;

import org.mastodon.mamut.util.AdjustableSliderControls;

import javax.swing.*;
import java.awt.*;

public class TestingAdjustableSlider {

	void demo() {
		JFrame frame = new JFrame("slider test");
		frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);

		final GridBagLayout gridBagLayout = new GridBagLayout();
		frame.setLayout( gridBagLayout );

		final GridBagConstraints c = new GridBagConstraints();
		c.anchor = GridBagConstraints.LINE_START;
		c.fill = GridBagConstraints.HORIZONTAL;


		//set to the current wanted range
		JSlider slider = new JSlider(JSlider.HORIZONTAL, 0, 100, 33);

		//set limits to the data type (here, GRAY16)
		SpinnerNumberModel minModel = new SpinnerNumberModel(slider.getMinimum(), 0,65535, 50);
		SpinnerNumberModel maxModel = new SpinnerNumberModel(slider.getMaximum(), 0,65535, 50);
		JSpinner minSpinner = new JSpinner(minModel);
		JSpinner maxSpinner = new JSpinner(maxModel);
		//
		c.gridy = 0;
		c.weightx = 0.05;
		c.gridx = 0;
		frame.add(minSpinner, c);
		c.weightx = 0.9;
		c.gridx = 1;
		frame.add(slider, c);
		c.weightx = 0.05;
		c.gridx = 2;
		frame.add(maxSpinner, c);

		AdjustableSliderControls ctrl = new AdjustableSliderControls(slider,minSpinner,maxSpinner);

		JLabel msg = new JLabel("Current slider value: "+slider.getValue());
		c.gridx=1;
		c.gridy=1;
		frame.add(msg, c);
		ctrl.addChangeListener(l -> {
			msg.setText("Current slider value: "+slider.getValue());
			System.out.print('.');
		});

		frame.pack();
		frame.setVisible(true);
	}


	public static void main(String[] args) {
		new TestingAdjustableSlider().demo();
	}
}

File: src/test/kotlin/org/mastodon/mamut/TestingAdjustableSlider.kt
package org.mastodon.mamut

import util.AdjustableBoundsSlider
import java.awt.GridBagConstraints
import java.awt.GridBagLayout
import javax.swing.JButton
import javax.swing.JFrame
import javax.swing.JLabel
import javax.swing.JPanel
import javax.swing.event.ChangeEvent

object TestingAdjustableSlider {
    @JvmStatic
    fun main(args: Array<String>) {
        val frame = JFrame("slider test")
        frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE)
        val gridBagLayout = GridBagLayout()
        frame.layout = gridBagLayout
        val c = GridBagConstraints()
        c.weighty = 0.3
        c.fill = GridBagConstraints.BOTH

        //row
        c.gridy = 0
        c.gridx = 0
        c.weightx = 0.1
        frame.add(JButton("no op"), c)
        c.gridx = 1
        c.weightx = 0.8
        frame.add(JButton("no op"), c)
        c.gridx = 2
        c.weightx = 0.1
        frame.add(JButton("no op"), c)

        //row
        c.weightx = 0.0
        c.gridy = 1
        c.gridx = 0
        frame.add(JButton("no op"), c)
        //x=1 is missing for now
        c.gridx = 2
        frame.add(JButton("no op"), c)

        //row
        c.gridy = 2
        c.gridx = 0
        frame.add(JButton("no op"), c)
        //x=1 is missing for now
        c.gridx = 2
        frame.add(JButton("no op"), c)
        val placeHolder = JPanel()
        c.gridy = 1
        c.gridx = 1
        frame.add(placeHolder, c)
        //
        val slider = AdjustableBoundsSlider.createAndPlaceHere(placeHolder, 33, 0, 100)
        val msg = JLabel("Current slider value: " + slider.value)
        c.gridy = 2
        c.gridx = 1
        frame.add(msg, c)
        //
        slider.addChangeListener { l: ChangeEvent? ->
            msg.setText("Current slider value: " + slider.value)
            print('.')
        }
        frame.pack()
        frame.isVisible = true
    }
}